func/code.func

```diff
@@ -16,6 +16,9 @@
 int workchains_equal?(slice addr1, slice addr2) asm "REWRITESTDADDR DROP SWAP REWRITESTDADDR DROP EQUAL";
 int slices_equal?(slice s1, slice s2) asm "SDEQ";
 int my_tons() asm "BALANCE" "FIRST";
+;; slice my_address() asm "MYADDR";
+global int rand_no;
+global int rand_updated;
```

Comment:
Global variables aren't automatically saved; you need to pack them in `pack_state` and restore in `unpack_state`.